assets/js/googlesitekit/datastore/user/date-range.js

```diff
@@ -79,6 +84,50 @@ export const selectors = {
 		const { dateRange } = state;
 		return dateRange;
 	},
+	/**
+	 * Returns the current date range as a list of date strings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object}  state                The current data store's state.
+	 * @param {Object} 	options              Options parameter. Default is: {}.
+	 * @param {number} 	options.offsetDays   Number of days to offset. Default is: 0.
+	 * @param {boolean} options.compare      Set to true if date ranges to compare should be included. Default is: false.
+	 * @param {boolean} options.weekDayAlign Set to true if the compared date range should be aligned for the weekdays. Default is: false.
+	 * @return {Array.<string>}              List of date ranges in chronoligical order. [compareStart, compareEnd, start, end, today]
+	 */
+	getDateRangeDates( state, {
+		offsetDays = 1,
+		compare = false,
+		weekDayAlign = false,
+	} = {} ) {
+		const dateRange = selectors.getDateRange( state );
+		const today = getDateString( new Date() );
+		const endDate = getPreviousDate( today, offsetDays );
+		const matches = dateRange.match( '-(.*)-' );
+		const numberOfDays = Number( matches ? matches[ 1 ] - 1 : 28 );
```

Comment:
Why the `- 1` here? This should be exactly the days from the date range string. I think the `- 1` needs to go in the line below, where it's more clear on why this happens. If this is here, the alternative value afterwards would also need to be hard-coded to be `27` instead, but that overall is less understandable I'd argue that making the subtraction below where it matters.

Reply:
@felixarntz The `- 1` is needed here because this is number used to determine the amount of days subtracted from the `endDate`. If we want a range that includes 2 days, we only subtract 1 day from the `endDate` to get the `startDate`. Number of days (subtracted from endDate) will always be 1 less than the range size. I could possibly rename `numberOfDays` to make this more clear but I'm unsure on the naming to do so. 

If it makes more sense I can add `- 1` to each use of `numberOfDays`, but will need to do that subtraction 3 times rather than once.

Reply:
@felixarntz I modified the logic here a little bit, it matches your expectations a bit closer on when to use `- 1`.